src/main/scala/it/pagopa/interop/agreementprocess/api/impl/AgreementApiServiceImpl.scala

```diff
@@ -806,4 +783,76 @@ final case class AgreementApiServiceImpl(
         .whenA(activeAgreement.nonEmpty)
     } yield ()
   }
+
+  override def addAgreementConsumerDocument(agreementId: String, documentSeed: DocumentSeed)(implicit
+    contexts: Seq[(String, String)],
+    toEntityMarshallerDocument: ToEntityMarshaller[Document],
+    toEntityMarshallerProblem: ToEntityMarshaller[Problem]
+  ): Route = authorize(ADMIN_ROLE) {
+
+    logger.info(s"Adding a consumer document to agreement $agreementId")
+
+    val result: Future[Document] = for {
+      agreementUUID <- agreementId.toFutureUUID
+      document      <- agreementManagementService.addConsumerDocument(
```

Comment:
🚓 we need to check that the requester is actually the agreement consumer (and maybe that the agreement is in a specific state? I guess it should not be possible to add documents on an active agreement?)